pkg/build/admission/admission.go

```diff
@@ -43,21 +43,57 @@ func NewBuildByStrategy(client client.Interface) admission.Interface {
 const (
 	buildsResource       = "builds"
 	buildConfigsResource = "buildconfigs"
+
+	cloneSubresource       = "clone"
+	instantiateSubresource = "instantiate"
 )
 
 func (a *buildByStrategy) Admit(attr admission.Attributes) error {
-	if resource := attr.GetResource(); resource != buildsResource && resource != buildConfigsResource {
-		return nil
-	}
-	var err error
-	switch obj := attr.GetObject().(type) {
-	case *buildapi.Build:
-		err = a.checkBuildAuthorization(obj, attr)
-	case *buildapi.BuildConfig:
-		err = a.checkBuildConfigAuthorization(obj, attr)
```

Comment:
> do we not just want a *buildapi.BuildRequest case with a checkBuildRequestAuthorization() func that does the build lookup and then calls checkBuildAuthorization?

The problem is that the meaning of BuildRequest.Name changes depending on whether it's sent with an bc/instantiate or a build/clone subresource. It's a bc name in the first case, build name in the second case.